### PR TITLE
name NEGs with ServicePort instead of TargetPort

### PR DESCRIPTION
**Pull request by agau4779**
Refactor that makes https://github.com/kubernetes/ingress-gce/pull/284 easier

### CODE REVIEW

File: pkg/utils/serviceport.go [hunk 1]
```diff
@@ -74,7 +75,7 @@ func (sp ServicePort) BackendName(namer *Namer) string {
 		return namer.IGBackend(sp.NodePort)
 	}
 
-	return namer.NEG(sp.ID.Service.Namespace, sp.ID.Service.Name, sp.SvcTargetPort)
+	return namer.NEG(sp.ID.Service.Namespace, sp.ID.Service.Name, fmt.Sprintf("%v", sp.SvcPort))
```
Comment (bowei) [inline pkg/utils/serviceport.go]: does it make sense to pass it as a string?

File: pkg/backends/backends_test.go [hunk 2]
```diff
@@ -857,7 +857,7 @@ func TestLinkBackendServiceToNEG(t *testing.T) {
 
 	for _, zone := range zones {
 		err := fakeNEG.CreateNetworkEndpointGroup(&computealpha.NetworkEndpointGroup{
-			Name: defaultNamer.NEG(namespace, name, port),
+			Name: defaultNamer.NEG(namespace, name, fmt.Sprintf("%v", svcPort.SvcPort)),
```
Comment (bowei) [inline pkg/backends/backends_test.go]: see comment below

**Comment by agau4779** [comment]
Incorporated into https://github.com/kubernetes/ingress-gce/pull/284 , addressed comments

**Pull request closed without merging**